cairo/scripts/zkpi_to_eels_json.py

```diff
@@ -0,0 +1,396 @@
+    script_dir : Path
+        Path to the script directory
+    """
+    print(f"Processing {zkpi_file.name}...")
+
+    # Load the ZKPI JSON
+    zkpi_data = load_zkpi_json(str(zkpi_file))
+
+    # Create code hash to code mapping
+    codes_mapping = create_code_hash_to_code_map(zkpi_data)
+
+    # Convert accounts using the code mapping
+    pre_state = convert_accounts(zkpi_data, codes_mapping)
+
+    load = Load("Cancun", "cancun")
+    # Adding partial state root
+    pre_state_root = state_root(load.json_to_state(pre_state))
+
+    # Generate genesis block header
+    parent_block = zkpi_data["ancestors"][0]
+    # Modify parent state root to be the state root of the pre-state (partial)
+    parent_block["stateRoot"] = "0x" + pre_state_root.hex()
+
+    # Convert block parameters
+    block_parameters = create_eels_block_parameters(zkpi_data["block"])
+    # We modify parent hash to be the hash of the modified parent block
+    block_parameters["blockHeader"]["parentHash"] = (
+        "0x" + keccak256(rlp.encode(load.json_to_header(parent_block))).hex()
+    )
+
+    # Create the final fixture format
+    fixture = {
+        "genesisBlockHeader": parent_block,
+        "newBlockParameters": block_parameters,
+        "pre": pre_state,
+        "chainId": zkpi_data["chainConfig"].get("chainId", 1),
+    }
+
+    # Sanity checks
+    block = Block(
+        header=load.json_to_header(fixture["newBlockParameters"]["blockHeader"]),
+        transactions=tuple(
+            (
+                LegacyTransaction(
+                    nonce=hex_to_u256(tx["nonce"]),
+                    gas_price=hex_to_uint(tx["gasPrice"]),
+                    gas=hex_to_uint(tx["gas"]),
+                    to=Address(hex_to_bytes(tx["to"])) if tx["to"] else Bytes0(),
+                    value=hex_to_u256(tx["value"]),
+                    data=Bytes(hex_to_bytes(tx["data"])),
+                    v=hex_to_u256(tx["v"]),
+                    r=hex_to_u256(tx["r"]),
+                    s=hex_to_u256(tx["s"]),
+                )
+                if isinstance(tx, dict)
+                else Bytes(hex_to_bytes(tx))
+            )  # Non-legacy txs are hex strings
+            for tx in fixture["newBlockParameters"]["transactions"]
+        ),
+        ommers=(),
+        withdrawals=tuple(
+            Withdrawal(
+                index=U64(int(w["index"], 16)),
+                validator_index=U64(int(w["validatorIndex"], 16)),
+                address=Address(hex_to_bytes(w["address"])),
+                amount=U256(int(w["amount"], 16)),
+            )
+            for w in fixture["newBlockParameters"]["withdrawals"]
+        ),
+    )
+
+    genesis_block = Block(
+        header=load.json_to_header(fixture["genesisBlockHeader"]),
+        transactions=(),
+        ommers=(),
+        withdrawals=(),
+    )
+
+    blockchain = BlockChain(
+        blocks=[genesis_block],
+        state=load.json_to_state(fixture["pre"]),
```

Comment:
note: when integrating with keth you will need to convert the state to a keth compatible one with defaultdict in tries (see ef-tests)